### DISCUSSION TITLE
Deserialize json to std::vector of objects

### DISCUSSION THREAD

**Discussion by NamelessGhost** [nlohmann/json#2694] [Q&A] [answered]
Hi there

I'm trying to save some settings using a json file. I was able to serialize my `std::vector<object> myVector` without to much trouble using `json myJson = myVector` and then I use `myJson.dump(1,9)` So far so good, stick with me.

When I try to **deserialize that json string back into myVector** using the below code I'm greeted with a "no match for operator=" error.

```
std::ifstream myFile;
myFile.open("myJsonFile");

auto jsonInput = json::parse(myFile);
myVector = jsonInput;   -- ERROR
```

Could you please give me a hint on how to do this the right way? I searched through every post I could find but was not able to figure it out myself.

Thanks a lot!
NamelessGhost

**Comment by nlohmann** [comment, marked as answer]
You need to provide a conversion from JSON to your `object` type. See https://github.com/nlohmann/json#arbitrary-types-conversions for more information.

**Reply by NamelessGhost** [reply]
I just figured it out the minute you wrote me. Stupid me forgot to implement a `from_json` conversion function. Thanks anyway, hope this is helpful for someone else. **This worked for me in the end**: 

`myVector = jsonInput.get<std::vector<MyObject>>();`

Btw. make sure your object is default constructible! This means that you don't have any parameters in the constructor that don't have a default value. Or just overload the constructor, which is the way I solved it. Look here:

`MyNS::MyConstructor(int param1, float param2)                // Doesn't work!`


`MyNS::MyConstructor(int param1 = 1, float param2 = 1)    // Works`


```
MyNS::MyConstructor(int param1, float param2)    // Works because...
MyNS::MyConstructor()                            // ... this one makes it work
```
